In [1]:
import scipy.io as sio
import numpy as np
import h5py as h5
import pandas as pd

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters


In [2]:
mat = "../cs274c-data/imdb.mat"

For matlab arrays before v7.3, use scipy.io.loadmat('file.mat')

For matlab arrays v7.3 and above, use h5py.File('file.mat','r')

To see contents of file.mat without loading into memory, use scipy.io.whosmat('file.mat')

In [3]:
sio.whosmat(mat)

[('imdb', (1, 1), 'struct')]

In [4]:
f = sio.loadmat(mat)

In [5]:
type(f)

dict

In [6]:
for key in f.keys():
    print(key)

__header__
__version__
__globals__
imdb


In [7]:
data = f['imdb'][0][0]
print(type(data))
print(len(data))

<class 'numpy.void'>
10


In [26]:
genders = data[3][0][:]
len(genders)
np.nansum(np.array(genders))

263214.0

In [8]:
data.dtype

dtype([('dob', 'O'), ('photo_taken', 'O'), ('full_path', 'O'), ('gender', 'O'), ('name', 'O'), ('face_location', 'O'), ('face_score', 'O'), ('second_face_score', 'O'), ('celeb_names', 'O'), ('celeb_id', 'O')])

In [9]:
for i in range(len(data)):
    print(data[i][0][5], data[i][0][5].dtype)

702986 int32
1991 uint16
['02/nm0000002_rm1075631616_1924-9-16_1991.jpg'] <U44
0.0 float64
['Lauren Bacall'] <U13
[[ 686.67682059  458.46721373 1198.10043604  969.89082917]] float64
3.4905630909496344 float64
1.3453352771625287 float64
['A.D. Miles'] <U10
11516 uint16


In [10]:
print(type(data[3][0]))
data[3][0]

<class 'numpy.ndarray'>


array([1., 1., 1., ..., 0., 0., 0.])

In [11]:
d = {'filename': data[2][0], 'gender': data[3][0]}

In [12]:
df = pd.DataFrame(data=d)

In [13]:
df.filename[0][0]

'01/nm0000001_rm124825600_1899-5-10_1968.jpg'

In [14]:
df['path'] = df.filename.apply(lambda x: x[0])

In [15]:
df.head()

filename  gender  \
0   [01/nm0000001_rm124825600_1899-5-10_1968.jpg]     1.0   
1  [01/nm0000001_rm3343756032_1899-5-10_1970.jpg]     1.0   
2   [01/nm0000001_rm577153792_1899-5-10_1968.jpg]     1.0   
3   [01/nm0000001_rm946909184_1899-5-10_1968.jpg]     1.0   
4   [01/nm0000001_rm980463616_1899-5-10_1968.jpg]     1.0   

                                           path  
0   01/nm0000001_rm124825600_1899-5-10_1968.jpg  
1  01/nm0000001_rm3343756032_1899-5-10_1970.jpg  
2   01/nm0000001_rm577153792_1899-5-10_1968.jpg  
3   01/nm0000001_rm946909184_1899-5-10_1968.jpg  
4   01/nm0000001_rm980463616_1899-5-10_1968.jpg

In [18]:
values, counts = np.unique(data[3], return_counts=True)

In [19]:
values

array([ 0.,  1., nan, ..., nan, nan, nan])

In [20]:
counts

array([189047, 263214,      1, ...,      1,      1,      1])

There are 189047 images of females

There are 263214 images of males